gdk4/src/texture.rs

```diff
@@ -18,3 +21,51 @@ impl<O: IsA<Texture>> TextureExtManual for O {
         }
     }
 }
+
+impl Texture {
+    pub fn for_pixbuf_future(
+        pixbuf: &'static gdk_pixbuf::Pixbuf,
+    ) -> Pin<Box<dyn Future<Output = Option<Texture>> + 'static>> {
+        Box::pin(async move {
+            let (sender, receiver) = futures_channel::oneshot::channel();
+
+            let sender = std::cell::Cell::new(Some(sender));
+
+            let task = gio::Task::new(
+                Some(&pixbuf.upcast_ref()),
```

Comment:
This should really take an `IsA<Object>` instead of `Object` (and then also use the same type for the callback)

Reply:
I don't follow

Reply:
The `gio::Task::new()` API should be improved so you don't need to cast that much